var/spack/repos/builtin/packages/rose/package.py

```diff
@@ -29,43 +29,56 @@
 from spack import *
 
 
-class Rose(Package):
+class Rose(AutotoolsPackage):
     """A compiler infrastructure to build source-to-source program
        transformation and analysis tools.
        (Developed at Lawrence Livermore National Lab)"""
 
     homepage = "http://rosecompiler.org/"
-    url = "https://github.com/rose-compiler/rose/archive/v0.9.7.tar.gz"
 
-    version('0.9.7', 'e14ce5250078df4b09f4f40559d46c75')
-    version('master', branch='master',
+    version('develop', branch='master',
+            git='https://github.com/rose-compiler/rose-develop.git')
+    version('0.9.7.0', commit='992c21ad06893bc1e9e7688afe0562eee0fda021',
             git='https://github.com/rose-compiler/rose.git')
-
-    patch('add_spack_compiler_recognition.patch')
+    version('0.9.9.0', commit='14d3ebdd7f83cbcc295e6ed45b45d2e9ed32b5ff',
+            git='https://github.com/rose-compiler/rose.git')
```

Comment:
Doesn't work with cached tarball:
```
configure: You are using a non-Git version of ROSE that lacks a VERSION file. This may be
configure: due to you having downloaded a ZIP archive of the ROSE repository from github.
configure: ROSE no longer supports being installed from a github ZIP archive; your ROSE
configure: source tree must contain Git version information, i.e., a ".git" directory.
```

Reply:
As indicated, ROSE can only be used with the source code from the Git repository. (Without the .git directory, the ROSE/VERSION file can't be generated.)

I've been using `spack purge --downloads && spack install rose`, but @tgamblin might suggest a better solution.

I know this isn't ideal, but it's currently the reality for ROSE until we finish our current work of generating binaries and proper source tarball releases.

Reply:
What is the `.git` directory used for?

Reply:
@tgamblin it's used to compute the ROSE version commit (otherwise stored in `ROSE/VERSION`), which is required to download the correct C/C++ frontend (EDG) binary during install.

The problem is, we don't generate `make dist` tarballs anymore otherwise we could simply download a proper ROSE release tarball that contains a pre-generated `VERSION` file with this same information.

So it's currently required to use the Git repository and not a tarball/zipball of the Git repository without the `.git` directory, which Spack is using as a cached copy.

Reply:
@justintoo: could you not add a custom `patch()` function to the Spack package to spit out a correct VERSION file for each version listed in the Spack package?

e.g. (notional versions):

```python
def patch(self):
    spec = self.spec
    with open('VERSION') as f:
        if '@0.9.9:' in spec:
            f.write('EDG 1.3')
        elif '@0.9.8:' in spec:
            f.write('EDG 1.1')
        elif '@0.9.7:' in spec:
            f.write('EDG 1.0')
```

etc.

Reply:
Also, is ROSE downloading the EDG front-end itself?  If so this isn't going to be mirrorable or installable on closed networks.  It needs to be a `resource` or a dependency to be properly mirrored.